cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -192,17 +193,34 @@ func Run(s *options.CMServer) error {
 		return err
 	}
 
-	// TODO: enable other lock types
-	rl := resourcelock.EndpointsLock{
-		EndpointsMeta: metav1.ObjectMeta{
-			Namespace: "kube-system",
-			Name:      "kube-controller-manager",
-		},
-		Client: leaderElectionClient,
-		LockConfig: resourcelock.ResourceLockConfig{
-			Identity:      id,
-			EventRecorder: recorder,
-		},
+	var rl resourcelock.Interface
+	switch s.LockType {
```

Comment:
Could we wrap this behind a utility function in leaderelection?  This will likely be a repeating pattern.